### ISSUE TITLE
Display link to attached PDF document, not a link to the conversation

### ISSUE THREAD

**Issue by Smile4ever** [jgeboski/purple-facebook#114]
If someone sends me a PDF document, I get something like this:
https://www.facebook.com/messages/17**50**26 [name_of_the_pdf.pdf]
instead of the direct link, like
https://cdn.fbsbx.com/hphotos-xpt1/v/t**.****-21/118**375_10****53449537229_1727378937_n.pdf/name_of_the_pdf.pdf?oh=5c99a6b**632afa**5b6d72**3e42b52&oe=55D6EAD2&dl=1

**Comment by dequis** [comment]
That's by design, unfortunately.

The plugin only shows direct links when there's a reasonable way to get them through the protocol, and in the case of attachments, we never get a direct link or even most of the components required to assemble the URL

This is what we get:

```
{
    "deltas": [
        {
            "deltaNewMessage": {
                "attachments": [
                    {
                        "fbid": 10207176710279837,
                        "fileSize": 4,
                        "filename": "virus.exe.pdf",
                        "id": "10207176710279837",
                        "mimeType": "application/pdf"
                    }
                ],
                "messageMetadata": {
                    "actorFbId": 134**460,
                    "messageId": "mid.1440003136087:0ca427ee25238b4993",
                    "offlineThreadingId": 6039810909602954791,
                    "tags": [
                        "source:chat:web"
                    ],
                    "threadKey": {
                        "otherUserFbId": 134**460
                    },
                    "timestamp": 1440003136452
                }
            }
        }
    ],
    "firstDeltaSeqId": 1094,
    "lastIssuedSeqId": 1094
}
```

And this is a corresponding link, copied from webchat:

`https://cdn.fbsbx.com/hphotos-xpt1/v/t59.2708-21/11408826_804538156319884_2072831910_n.pdf/virus.exe.pdf?oh=7ba31f9bd6c3027959a8e877f9749921&oe=55D66A4D&dl=1`

Note how we don't get *any* parts of that link in the above information.

In these cases, the best way we have to show that link is to point the user at the conversation URL, so they can grab it from there. Videos work the same way.

So I'm afraid i'll have to close this as wontfix, unless @jgeboski disagrees with me and thinks this is somehow possible. Sorry!

**Comment by marsjaninzmarsa** [comment]
5 minutes of googling, but I don't know if it's still working.

http://stackoverflow.com/questions/9192430/view-attachments-in-threads

**Comment by jgeboski** [comment]
This has been implemented as of [2d832af0f400](https://hg.pidgin.im/soc/2015/jgeboski/facebook/rev/2d832af0f400), and will appear in the next release.

@marsjaninzmarsa Thanks for the link!  This is still active and used.

**Comment by marsjaninzmarsa** [comment]
:smile_cat:
